### PR TITLE
Alternative <= function

### LINKED ISSUE #196: Refactor `<=`
Now that `AbstractPolygon.jl` exists, we could refactor `<=` into that file.

### PR DESCRIPTION
**Pull request by mforets**
Closes #196.

### CODE REVIEW

File: src/AbstractPolygon.jl [hunk 1]
```diff
@@ -44,3 +47,127 @@ The ambient dimension of the polygon, which is 2.
...
+direction (1, 0).
+
+### Algorithm
+
+The implementation checks the quadrant of each direction, and compares directions
+using the right-hand rule. In particular, it doesn't use the arctangent.
+"""
+function <=(u::AbstractVector{N}, v::AbstractVector{N})::Bool where {N<:Real}
+    qu, qv = quadrant(u), quadrant(v)
+    if qu - qv == 0
```
Comment (schillic) [inline src/AbstractPolygon.jl]: simpler: `qu == qv`

File: src/AbstractPolygon.jl [hunk 2]
```diff
@@ -44,3 +47,127 @@ The ambient dimension of the polygon, which is 2.
...
+
+     ^
+   1 | 0
+  ---+-->
+   2 | 3
+
+### Algorithm
+
+This function is inspired from AGPX's answer for C++ in the stackoverflow question
+[Sort points in clockwise order?](https://stackoverflow.com/questions/6989100/sort-points-in-clockwise-order).
```
Comment (schillic) [inline src/AbstractPolygon.jl]: You can refer to the exact post by clicking the "share" button (it is quite hidden):
https://stackoverflow.com/questions/6989100/sort-points-in-clockwise-order/46635372#46635372

File: src/AbstractPolygon.jl [hunk 3]
```diff
@@ -44,3 +47,127 @@ The ambient dimension of the polygon, which is 2.
...
+The idea is to encode the following logic function:
+`11 -> 0, 01 -> 1, 00 -> 2, 10 -> 3`, according to the convention of above.
+"""
+function quadrant(w::AbstractVector{N})::Int where {N<:Real}
+    dwx = w[1] >= zero(N) ? 1 : 0
+    dwy = w[2] >= zero(N) ? 1 : 0
+    qw = (1 - dwx) + (1 - dwy) + ((dwx & (1 - dwy)) << 1)
+    return qw
```
Comment (schillic) [inline src/AbstractPolygon.jl]: return immediately?

File: test/unit_Polygon.jl [hunk 4]
```diff
@@ -107,3 +107,23 @@ for N in [Float64, Float32] # TODO Rational{Int}
...
+v4 = N[-1.0, 1.0]
+v5 = N[-1.0, 0.0]
+v6 = N[-1.0, -1.0]
+v7 = N[0, -1.0]
+v8 = N[1.0, -1.0]
+v = [v1, v2, v3, v4, v5, v6, v7, v8]
+
+for (i, vi) in enumerate(v)
+    for j in 1:8
+        if i >= j
```
Comment (schillic) [inline test/unit_Polygon.jl]: simpler: `for j in 1:i`

File: src/AbstractPolygon.jl [hunk 5]
```diff
@@ -1,6 +1,9 @@
 export AbstractPolygon,
        tohrep,
-       tovrep
+       tovrep,
+       jump2pi
```
Comment (schillic) [inline src/AbstractPolygon.jl]: Do we actually have to export `jump2pi`?

File: docs/src/lib/representations.md [hunk 6]
```diff
@@ -184,6 +184,14 @@ tohrep(::VPolygon{Float64})
 tovrep(::VPolygon{Float64})
 ```
 
+### Sorting directions
+
+```@docs
+jump2pi
+<=(::AbstractVector{Float64}, ::AbstractVector{Float64})
```
Comment (schillic) [inline docs/src/lib/representations.md]: Do we also want to document the rational version?

File: test/unit_Polygon.jl [hunk 7]
```diff
@@ -107,3 +107,23 @@ for N in [Float64, Float32] # TODO Rational{Int}
     @test ⊆(HPolygon{N}(), p1)
     @test ⊆(p1, BallInf(N[1., 0.], N(1.)))
 end
+
+# Test comparison of directions <= for rational entries
```
Comment (schillic) [inline test/unit_Polygon.jl]: We should put this whole block inside the above for loop after adding `Rational` to its arguments.

**Comment by schillic** [comment]
Do we want to add `Rational{Int}` to the unit tests now or in a separate PR?

**Comment by schillic** [comment]
Oops, I sent two reviews, now some comments are marked as outdated. Please do not ignore them.

File: src/AbstractPolygon.jl (hunk 2 above)
Comment (mforets) [reply src/AbstractPolygon.jl]: good finding!

File: docs/src/lib/representations.md (hunk 6 above)
Comment (mforets) [reply docs/src/lib/representations.md]: actually it gets documented with this line (don't ask why).

File: src/AbstractPolygon.jl (hunk 5 above)
Comment (mforets) [reply src/AbstractPolygon.jl]: not really, let's take it out.

File: docs/src/lib/representations.md (hunk 6 above)
Comment (schillic) [reply docs/src/lib/representations.md]: okay... :)

File: test/unit_Polygon.jl (hunk 4 above)
Comment (mforets) [reply test/unit_Polygon.jl]: ok let's merge them (though i have to swap the comparison)

Comment (schillic) [reply test/unit_Polygon.jl]: really?

**Comment by mforets** [comment]
> Do we want to add Rational{Int} to the unit tests now or in a separate PR?

i tried to include it but 

```julia
  while loading /Users/forets/.julia/v0.6/LazySets/test/unit_Polygon.jl, in expression starting on line 1
Test Summary:    | Pass  Error  Total
LazySets.Polygon |  232      1    233
ERROR: LoadError: Some tests did not pass: 232 passed, 0 failed, 1 errored, 0 broken.
while loading /Users/forets/.julia/v0.6/LazySets/test/runtests.jl, in expression starting on line 14
```

==> separate PR

File: test/unit_Polygon.jl [hunk 8]
```diff
@@ -106,8 +106,26 @@ for N in [Float64, Float32] # TODO Rational{Int}
...
+    v3 = N[0.0, 1.0]
+    v4 = N[-1.0, 1.0]
+    v5 = N[-1.0, 0.0]
+    v6 = N[-1.0, -1.0]
+    v7 = N[0, -1.0]
+    v8 = N[1.0, -1.0]
+    v = [v1, v2, v3, v4, v5, v6, v7, v8]
+
+    for (i, vi) in enumerate(v)
+        for j in i:8
```
Comment (schillic) [inline test/unit_Polygon.jl]: `1:i`, right? (same below)

Comment (mforets) [reply test/unit_Polygon.jl]: for these definitions of `vi`, we should check that `v1 <= v1`, `v1 <= v2`, `v1 <= v3`, and so on until `v1 <= v8`. similarly `v2 <= v2`, until `v2 <= v8`. `v8` is the greatest angle thus it is only tested against itself.

Comment (schillic) [reply test/unit_Polygon.jl]: Oh, I read `i` as `1`, is see now what you did.

**Pull request merged**